### PR TITLE
Copy Bragi2 error handling 

### PR DESCRIPTION
**Pull request by AzHicham**

### CODE REVIEW

File: libs/mimir/src/adapters/primary/bragi/routes.rs [hunk 1]
```diff
@@ -199,6 +209,7 @@ pub async fn ensure_lat_lon_consistent(
     if params.lat.is_some() ^ params.lon.is_some() {
         Err(warp::reject::custom(InvalidRequest {
             reason: InvalidRequestReason::InconsistentLatLonRequest,
+            info: "".to_string(),
```
Comment (crocme10) [inline libs/mimir/src/adapters/primary/bragi/routes.rs:212]: That's not very informative...

File: libs/mimir/src/adapters/primary/bragi/routes.rs [hunk 2]
```diff
@@ -223,6 +234,7 @@ pub async fn ensure_zone_type_consistent(
     {
         Err(warp::reject::custom(InvalidRequest {
             reason: InvalidRequestReason::InconsistentZoneRequest,
+            info: "".to_string(),
```
Comment (crocme10) [inline libs/mimir/src/adapters/primary/bragi/routes.rs:237]: ditto

File: libs/mimir/src/adapters/primary/bragi/routes.rs [hunk 3]
```diff
@@ -276,19 +290,33 @@ pub fn status() -> impl Filter<Extract = (), Error = Rejection> + Clone {
...
     } else {
-        // Do better error handling here
-        Ok(warp::reply::with_status(
-            reply,
-            StatusCode::INTERNAL_SERVER_ERROR,
-        ))
-    }
+        warp::reply::with_status(
+            warp::reply::json(&ApiError {
+                short: "validation error".to_string(),
```
Comment (crocme10) [inline libs/mimir/src/adapters/primary/bragi/routes.rs:312]: same "short" as  InvalidRequest

File: libs/mimir/src/adapters/primary/bragi/routes.rs [hunk 4]
```diff
@@ -276,19 +290,33 @@ pub fn status() -> impl Filter<Extract = (), Error = Rejection> + Clone {
...
+                long: "fsdfsdf".to_string(),
```
Comment (crocme10) [inline libs/mimir/src/adapters/primary/bragi/routes.rs:313]: Lost in translation

File: libs/mimir/src/adapters/primary/bragi/routes.rs (hunk 1 above)
Comment (AzHicham) [reply libs/mimir/src/adapters/primary/bragi/routes.rs:212]: Well this is a Work In Progress so…
it’s a draft PR, I’ll add you to reviewer list when job is done !

File: libs/mimir/src/adapters/primary/bragi/routes.rs [hunk 5]
```diff
@@ -196,12 +209,28 @@ pub async fn ensure_query_string_not_empty(
...
     params: ForwardGeocoderQuery,
 ) -> Result<ForwardGeocoderQuery, Rejection> {
-    if params.lat.is_some() ^ params.lon.is_some() {
-        Err(warp::reject::custom(InvalidRequest {
+    match (params.lat, params.lon) {
+        (Some(lat), Some(lon)) => {
+            if !(-90f32..=90f32).contains(&lat) {
+                Err(warp::reject::custom(InvalidRequest {
+                    reason: InvalidRequestReason::OutOfRangeLatLonRequest,
+                    info: "latitude parameter is outside of range [-90;90]".to_string(),
```
Comment (pbench) [inline libs/mimir/src/adapters/primary/bragi/routes.rs:217]:
```suggestion
                    info: format!("requested latitude {} is outside of range [-90;90]", lat),
```

File: libs/mimir/src/adapters/primary/bragi/routes.rs [hunk 6]
```diff
@@ -196,12 +209,28 @@ pub async fn ensure_query_string_not_empty(
...
+                }))
+            } else if !(-180f32..=180f32).contains(&lon) {
+                Err(warp::reject::custom(InvalidRequest {
+                    reason: InvalidRequestReason::OutOfRangeLatLonRequest,
+                    info: "longitude parameter is outside of range [-180;180]".to_string(),
```
Comment (pbench) [inline libs/mimir/src/adapters/primary/bragi/routes.rs:222]:
```suggestion
                    info: format!("requested longitude {} is outside of range [-180;180]", lon),
```

**Review by pbench — APPROVED** [review]

**Review by azime — APPROVED** [review]

**Pull request merged**